bot/gaming/gaming.py

```diff
@@ -0,0 +1,168 @@
+        overwrites_text = self.cat_gaming_channels.overwrites
+        overwrites_text[ctx.author] = discord.PermissionOverwrite(manage_messages=True)
+        await self.cat_gaming_channels.create_text_channel(name=name, topic=topic, overwrites=overwrites_text,
+                                                           reason=reason)
+
+        log.info("Temporary Game Room created by %s", ctx.author)
+        await ctx.send(":white_check_mark: Der Game-Room wurde erfolgreich erstellt!",
+                       delete_after=const.TIMEOUT_INFORMATION)
+
+    @create_gaming_room.error
+    async def gaming_room_error(self, ctx: commands.Context, error: commands.CommandError):
+        """Error Handler for the command `gameroom`.
+
+        Handles the exceptions which could occur during the execution of this command.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            error (commands.CommandError): The error raised during the execution of the command.
+        """
+        if isinstance(error, commands.CommandInvokeError) and isinstance(error.original, NotImplementedError):
+            await ctx.send("Bitte lösche zuerst deinen bestehenden Game Room, bevor du einen neuen erstellst.",
+                           delete_after=const.TIMEOUT_INFORMATION)
+        elif isinstance(error, commands.CommandInvokeError) and isinstance(error.original, RuntimeWarning):
+            await ctx.send("Es gibt zurzeit zu viele aktive Game Rooms. Bitte versuche es später noch einmal. "
+                           ":hourglass:", delete_after=const.TIMEOUT_INFORMATION)
+        elif isinstance(error, commands.CommandInvokeError) and isinstance(error.original, discord.InvalidArgument):
+            await ctx.send("Das Nutzer-Limit für einen Sprachkanal darf 99 nicht überschreiten. Bitte versuche es noch "
+                           "einmal.", delete_after=const.TIMEOUT_INFORMATION)
+
+    def _determine_channel_number(self, name: str) -> Optional[str]:
+        """Method which creates a string representing a channel number if needed.
+
+        If a member tries to create a Game Room with a name that is currently being used, this method will determine
+        the next free numeration and returns a corresponding string which will then be added to the channel name.
+        If the name chosen is unique, `None` will be returned instead.
+
+        Args:
+            name (str): The channel name chosen by the member.
+
+        Returns:
+            Optional[str]: A string representing the next free numeration for a channel with the specified name.
+        """
+        channels = self.cat_gaming_channels.voice_channels
+
+        try:
+            existing_name = next(ch.name for ch in reversed(channels) if name in ch.name)
+            regex = re.search(r"\[#(\d+)]", existing_name)  # Regex for getting channel number.
```

Comment:
What happens if the user has the number in the name? or in his username?

Reply:
This will work too. The only "problem" would be if someone names his channel like the part before the "[#X]", then SAM will just add a channel number for this too which is 1 higher than the number in the other channel name.

Reply:
I think you should fix that. Adapt the regex to accept no further symbol after the numbers or make sth like a endswith check or so. It might not be very probable but it's still a bug